include/envoy/registry/registry.h

```diff
@@ -139,6 +160,19 @@ template <class Base> class FactoryRegistry {
     }
   }
 
+  /**
+   * Permanently disables the named factory by setting the corresponding
+   * factory pointer to null.
+   */
+  static bool disableFactory(absl::string_view name) {
+    auto it = factories().find(name);
+    if (it == factories().end()) {
+      return false;
+    }
+    it->second = nullptr;
```

Comment:
Can we just erase the name from `factories()`?

Reply:
I avoided that because there's no locking around the lookup. That should not be a problem in practice with the current usage, but it makes me nervous :)

Reply:
Changing the value should make you as nervous as erase :) because they are not thread safe either.

Reply:
IIUC it's safe to change the pointer. It has no effect on concurrent lookups and and caller either gets the pointer or gets null. Since we don't deallocate the factory, either is OK. It's racy but not unsafe.